src/librustdoc/doctest/make.rs

```diff
@@ -17,242 +17,392 @@ use rustc_span::FileName;
 
 use super::GlobalTestOptions;
 
-/// Transforms a test into code that can be compiled into a Rust binary, and returns the number of
-/// lines before the test code begins as well as if the output stream supports colors or not.
-pub(crate) fn make_test(
-    s: &str,
-    crate_name: Option<&str>,
-    dont_insert_main: bool,
-    opts: &GlobalTestOptions,
-    edition: Edition,
-    test_id: Option<&str>,
-) -> (String, usize, bool) {
-    let (crate_attrs, everything_else, crates) = partition_source(s, edition);
-    let everything_else = everything_else.trim();
-    let mut line_offset = 0;
-    let mut prog = String::new();
-    let mut supports_color = false;
-
-    if opts.attrs.is_empty() {
-        // If there aren't any attributes supplied by #![doc(test(attr(...)))], then allow some
-        // lints that are commonly triggered in doctests. The crate-level test attributes are
-        // commonly used to make tests fail in case they trigger warnings, so having this there in
-        // that case may cause some tests to pass when they shouldn't have.
-        prog.push_str("#![allow(unused)]\n");
-        line_offset += 1;
-    }
+pub(crate) struct DocTest {
```

Comment:
Why do we need this in addition to `RunnableDoctest`? And if it is necessary, can you give it a more descriptive name?

Reply:
`RunnableDoctest` contains information to run a doctest whereas `DocTest` contains information to generate a doctest.

Reply:
:+1:

Reply:
Yeah, that sounds like a good name :+1: